src/lib/assetBridger/erc20Bridger.ts

```diff
@@ -338,18 +348,38 @@ export class Erc20Bridger extends AssetBridger<
   /**
    * Get the corresponding L1 for the provided L2 token
    * @param erc20L1Address
-   * @param l1Provider
+   * @param l2Provider
+   * @param l1Provider Used to check that the returned l1 address is indeed registered with the L1 gateway
    * @returns
    */
   public async getL1ERC20Address(
     erc20L2Address: string,
-    l2Provider: Provider
+    l2Provider: Provider,
+    l1Provider?: Provider
   ): Promise<string> {
     await this.checkL2Network(l2Provider)
 
     const arbERC20 = L2GatewayToken__factory.connect(erc20L2Address, l2Provider)
+    const l1Address = await arbERC20.functions.l1Address().then(([res]) => res)
+
+    // @deprecated: for the next breaking change make the l1Provider required and always do the check below
+    if (l1Provider) {
```

Comment:
we can instead just check against the L2 router using the L2 provider

Reply:
good point. updated